### PR TITLE
Fixing tests for django 2.2

### PR DESCRIPTION
**Pull request by jonathan-s**
This PR gives us an overview of the current failing tests for cms4. Due to the merge of https://github.com/divio/django-cms/pull/6486

Publishing has been removed from core-cms4. Hence nothing to publish. Therefore deletion of that in tests.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -13,12 +13,13 @@ deps =
 
     dj111: Django>=1.11,<2.0
     dj20: Django>=2.0,<2.1
+    # Frontend tests fail after dj2.2.2 because of phantomjs and the change
```
Comment (Aiky30) [inline tox.ini]: We will need @FinalAngel and @vxsx to help us decide on the best solution for this I think.

File: djangocms_text_ckeditor/tests/test_widget.py [hunk 2]
```diff
@@ -36,7 +38,6 @@ def test_sub_plugin_config(self):
     def test_plugin_edit(self):
         page = create_page(title='pagina', template='page.html', language='en')
         add_plugin(page.get_placeholders('en').get(slot='content'), 'TextPlugin', 'en', body="Lorem ipsum")
-        page.publish('en')
```
Comment (Aiky30) [inline djangocms_text_ckeditor/tests/test_widget.py]: Can you add a description to the PR intro that states that publishing has been removed from the core of Django CMS v4?

Comment (jonathan-s) [reply djangocms_text_ckeditor/tests/test_widget.py]: Done.

File: tox.ini (hunk 1 above)
Comment (jonathan-s) [reply tox.ini]: To clarify; it works if you test things manually. It's just that phantomjs / casper is outdated.

Comment (vxsx) [reply tox.ini]: it fails most likely due to how clicks are implemented in casperjs/phantomjs, so short term it requires adaptions in the tests to account for the change in django, but best solution would be to move away from casperjs to something puppeteer based or smth like cypress

django cms would also need to do that eventually, but it's likely a much larger effort

i'm ok with disabling the faulty test for now

Comment (jonathan-s) [reply tox.ini]: Hi @vxsx I was never able to fully understand how you disable tests in casperjs. I mean you can comment out the test, but is there another way?

+1 for cypress.

Comment (vxsx) [reply tox.ini]: hi!
yeah there's no other way that i know of

Comment (Aiky30) [reply tox.ini]: @jonathan-s You had mentioned that you wanted to use Cypress so I guess this is music. Can we disable or comment out the failing test to unblock our current issue and then look at rewriting to use Cypress? Can be an ongoing task if @vxsx agrees.

Comment (jonathan-s) [reply tox.ini]: @vxsx I have now disabled the offending test by commenting it out.

Comment (vxsx) [reply tox.ini]: 👍

**Comment by jonathan-s** [comment]
@vxsx Thank you for approving this PR, is there anything that prevents this PR from getting merged?

**Comment by FinalAngel** [comment]
@jonathan-s not from our end, okay for you to merge as well?

**Comment by jonathan-s** [comment]
@jonathan-s Please do merge! We are happy with this.

File: tox.ini [hunk 3]
```diff
@@ -13,12 +13,11 @@ deps =
 
     dj111: Django>=1.11,<2.0
     dj20: Django>=2.0,<2.1
+    dj22: Django>=2.2,<3.0
 
     cms40: https://github.com/divio/django-cms/archive/release/4.0.x.zip
 
 basepython =
-    py27: python2.7
```
Comment (Aiky30) [inline tox.ini]: We will need to get the removal of 2.7 agreed by Divio, it's not our call to remove this.

File: .travis.yml [hunk 4]
```diff
@@ -6,12 +6,12 @@ sudo: false
 
 python:
   - 3.5
-  - 2.7
```
Comment (Aiky30) [inline .travis.yml]: May need to add this back in, depends what Divio state for this as the CMS is still 2.7 compatible.

File: .travis.yml [hunk 5]
```diff
@@ -24,9 +24,6 @@ before_script:
 # command to install dependencies, e.g. pip install -r requirements.txt --use-mirrors
 install:
   - pip install -U tox>=1.8 coveralls
-  - "if [[ $TRAVIS_PYTHON_VERSION == '2.7' ]]; then export PYVER=py27; fi"
```
Comment (Aiky30) [inline .travis.yml]: Need to get this agreed.

**Pull request merged**